src/editor/shadow-taxonomies/parent-listings.js

```diff
@@ -0,0 +1,357 @@
+/**
+ * UI for managing parent listings.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf, _x } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import {
+	Button,
+	ExternalLink,
+	Modal,
+	Notice,
+	PanelRow,
+	Spinner,
+	ToggleControl,
+} from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { PluginDocumentSettingPanel } from '@wordpress/edit-post';
+import { useEffect, useState } from '@wordpress/element';
+import { decodeEntities } from '@wordpress/html-entities';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * External dependencies
+ */
+import { AutocompleteWithSuggestions } from 'newspack-components';
+
+/**
+ * Internal dependencies
+ */
+import { getPostTypeByTaxonomy, getTaxonomyLabel, isListing } from '../utils';
+import './style.scss';
+
+const ParentListingsComponent = ( { hideParents, postId, updateMetaValue } ) => {
+	const [ isUpdating, setIsUpdating ] = useState( false );
+	const [ modalVisible, setModalVisible ] = useState( false );
+	const [ initialParentTerms, setInitialParentTerms ] = useState( [] );
+	const [ parentTerms, setParentTerms ] = useState( [] );
+	const [ selectedPostType, setSelectedPostType ] = useState( null );
+	const [ message, setMessage ] = useState( null );
+	const postType = window?.newspack_listings_data.post_type;
+	const postTypes = window?.newspack_listings_data.post_types || {};
+	const taxonomies = window?.newspack_listings_data.taxonomies || {};
+
+	// Fetch suggestions for suggestions list on component mount.
+	useEffect(() => {
+		setMessage( null );
+		apiFetch( {
+			path: addQueryArgs( '/newspack-listings/v1/parents', {
+				per_page: 100,
+				post_id: postId,
+			} ),
+		} )
+			.then( response => {
+				const mappedResponse = response.map( post => ( {
+					value: post.value,
+					label: post.label,
+					postType: post.post_type,
+				} ) );
+				setParentTerms( mappedResponse );
+				setInitialParentTerms( mappedResponse );
+			} )
+			.catch( e => {
+				setMessage( {
+					status: 'error',
+					children: e.message || __( 'Error fetching suggestions.', 'newspack-listings' ),
+					isDismissible: false,
+				} );
+			} );
+	}, []);
+
+	// Determine whether the current post can have parent listings.
+	let canHaveParents = false;
+	for ( const taxonomy in taxonomies ) {
+		const taxonomyPostType = getPostTypeByTaxonomy( taxonomies[ taxonomy ].name );
+		const postTypeIsChild = -1 < taxonomies[ taxonomy ].post_types.indexOf( postType );
+
+		// Posts, pages, and listings can have parents if they can be assigned listing shadow terms.
+		if ( postTypeIsChild && taxonomyPostType !== postType ) {
+			canHaveParents = true;
+		}
+	}
+
+	// Bail early if the post type can't have parent listings.
+	if ( ! canHaveParents || isListing() ) {
+		return null;
```

Comment:
Same as in `child-listings.js` - the initial fetch should not happen if the component will return `null`.

Reply:
Fixed in 1902d9d.